### PR TITLE
SOLR-17473: Introduce build tools formatting

### PR DESCRIPTION
**Pull request by malliaridis**
https://issues.apache.org/jira/browse/SOLR-17473

# Description

With the build tools persisting of various Java files, formatting and checking tools should also be run on these files.

# Solution

This PR migrates the `buildSrc` to composite build project, as demonstrated in https://github.com/apache/lucene/pull/13484 and applied in https://github.com/apache/solr/pull/2706 (with minor adjustments).

# Tests

No tests have been added or altered.

# Checklist

Please review the following and check all that apply:

- [X] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [X] I have created a Jira issue and added the issue ID to my pull request title.
- [X] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended, not available for branches on forks living under an organisation)
- [X] I have developed this patch against the `main` branch.
- [X] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Review by epugh — COMMENTED** [review]
I had no idea we had so many .java files related to doing our builds.   I am always in favour of having more code strictness in an environment where many cooks are in the kitchen!

**Comment by epugh** [comment]
^ @malliaridis  I triggered the github build and looks like something burped.

**Review by dsmiley — COMMENTED** [review]
Since some of the build infra source files come from Lucene in their entirety, why not leave them be in the Lucene namespace?  Lucene & Solr continue to have a build that's rather aligned, allowing taking of changes from one project to the other.

**Comment by malliaridis** [comment]
@dsmiley I thought the naming was just a legacy thing in this case, but for just this case I made a separate commit for the renaming, so that I can undo it easily. 

@epugh thanks for executing the pipeline. The error seems not related to the changes though. Any idea what could be the cause?

**Review by dsmiley — APPROVED** [review]
I did another short look in a hurry.  If I had more time, I might compare to Lucene project but I don't.  +1 approval from me.  I'll merge tomorrow but maybe sooner with someone else's +1.

File: build-tools/build-infra/build.gradle [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  targetCompatibility = scriptDepVersions['min-java-version']
+}
+
+gradlePlugin {
+  automatedPublishing = false
+
+  plugins {
+    buildInfra {
+      id = 'solr.build-infra'
+      implementationClass = 'org.apache.lucene.gradle.buildinfra.BuildInfraPlugin'
```
Comment (iamsanjay) [inline build-tools/build-infra/build.gradle:51]: Can we change it to `org.apache.solr`?

Comment (dsmiley) [reply build-tools/build-infra/build.gradle:51]: See my other feedback in this PR on that topic.

**Pull request merged**